luxtronik/calculations.py

```diff
@@ -122,7 +123,16 @@ def __init__(self):
             78: HeatpumpCode("ID_WEB_Code_WP_akt"),
             79: BivalenceLevel("ID_WEB_BIV_Stufe_akt"),
             80: OperationMode("ID_WEB_WP_BZ_akt"),
-            81: Version("ID_WEB_SoftStand"),
+            81: version,
+            82: version.spawn_child(),
+            83: version.spawn_child(),
+            84: version.spawn_child(),
+            85: version.spawn_child(),
+            86: version.spawn_child(),
+            87: version.spawn_child(),
+            88: version.spawn_child(),
+            89: version.spawn_child(),
+            90: version.spawn_child(),
```

Comment:
- Does this datatype no longer have a name (`ID_WEB_SoftStand`)?
- I totally get that the handling of this datatype was not particularly nice and iterating over it was generating confusing results. While this fixes it, the definition doesn't look super handy. Couldn't we address this in the iterator itself somehow by indicating the length of each datatype, etc.?

Reply:
* The name is unchanged, it is defined in Line 43.
* The version datatype is the only one with a length bigger than 1. Thus, it was more reasonable to me to handle it as an exceptional case by introducing the "sub"-datatype.
* To which "iterator" are you referring to?

Reply:
> The name is unchanged, it is defined in Line 43.

Ah, somehow missed that on first glance, sorry for that.

> The version datatype is the only one with a length bigger than 1. Thus, it was more reasonable to me to handle it as an exceptional case by introducing the "sub"-datatype.

Depending on how we're going to tackle #33, we might have other datatypes with length > 1, also.

Personally I would prefer some kind of length attribute (with the default being 1), compared to this approach.

> To which "iterator" are you referring to?

In the description of this pull request you were referring to the iterator. I suppose you mean the `__iter__` method of `Calculations`. Haven't checked this in close detail yet, probably it didn't need to skip anything before, since the values have not been considered for iteration in the first place.

Reply:
Concerning #33: In my opinion, the data there has length one (each time sits in one slot). The overall handling of the data there should be (in my opinion) handled in a higher-level way in the sense of #119.